app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -55,6 +56,23 @@ public void clear() {
         notifyDataSetChanged();
     }
 
+    // uses basic search algorithm
+    public int search(String id){
+        for (int index = 0; index < getItemCount(); index++){
+            Show s = shows.get(index);
+            if(s.getId().equals(id)){
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    public void updateShowImage(int i, String imgPath) {
+        Show show = shows.get(i);
+        show.setImageUrl(imgPath);
+        notifyItemChanged(i);
+    }
```

Comment:
Hm, not sure f this makes sense here -- this is only updating the list here, right?  But you also have this list in the fragment -- I think you might want to do this in the fragment to ensure that list is up to date, and then just tell the adapter that the item changed at the specific index